src/oop_ext/foundation/types_.py

```diff
@@ -87,47 +90,46 @@ def __getattr__(self, mname):
 
         return self
 
-    def __setattr__(self, _name, _value):
+    def __setattr__(self, _name: str, _value: object) -> Any:
         "Ignore attribute setting."
         return self
 
-    def __delattr__(self, _name):
+    def __delattr__(self, _name: str) -> None:
         "Ignore deleting attributes."
-        return self
 
-    def __enter__(self):
+    def __enter__(self) -> Any:
```

Comment:
I may be mistaken, but wouldn't this be a case of using TypeVar?

```python
T = TypeVar('T')

from typing import TypeVar

def __enter__(self: T) -> T:

Reply:
Fixed!